### PR TITLE
Tutorial+petting zoo

### PR DESCRIPTION
**Pull request by jyan1999**
# Description

Added a tutorial notebook 

Added PettingZoo environment

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by micahcarroll — APPROVED** [review]
Thanks so much @jyan1999!! 💯  Pending on whether @elliottower thinks this is sufficient to make the environment compatible with PettingZoo, we can merge this!

File: src/overcooked_ai_py/mdp/overcooked_env.py [hunk 1]
```diff
@@ -664,6 +665,136 @@ def proportion_stuck_time(trajectories, agent_idx, stuck_time=3):
...
+    """
+    Create environment for gymnasium for pettingZoo
+    """
+
+    def step(self, joint_action):
+        obs, reward, done, info = super().step(joint_action)
+        # from the doc, this is true only if a pre set goal is achieved, which we don't have in this environment
+        # the only termination condition is the preset max steps in the environment
+        terminated = False
+        truncated = done
```
Comment (elliottower) [inline src/overcooked_ai_py/mdp/overcooked_env.py:678]: Is the done signal sent only when a time limit is reached? Usually I’d expect terminated to be set to done and truncated to be determined by a time limit

File: src/overcooked_ai_py/mdp/overcooked_env.py [hunk 2]
```diff
@@ -664,6 +665,136 @@ def proportion_stuck_time(trajectories, agent_idx, stuck_time=3):
...
+        return obs, reward, terminated, truncated, info
+
+
+from pettingzoo.utils.env import ParallelEnv
+
+from overcooked_ai_py.agents.agent import AgentPair
+
+
+class OvercookedEnv_Pet(ParallelEnv):
```
Comment (elliottower) [inline src/overcooked_ai_py/mdp/overcooked_env.py:687]: Nitpick but python style guides generally say not to use underscores in class names (or to mix camel case and underscores)

File: src/overcooked_ai_py/mdp/overcooked_env.py [hunk 3]
```diff
@@ -664,6 +665,136 @@ def proportion_stuck_time(trajectories, agent_idx, stuck_time=3):
...
+    def __init__(self, base_env, agents):
+        """
+        base_env: OvercookedEnv
+        featurize_fn(mdp, state): fn used to featurize states returned in the 'both_agent_obs' field
+
+        Example creating a gym env:
+
+        mdp = OvercookedGridworld.from_layout_name("asymmetric_advantages")
+        base_env = OvercookedEnv.from_mdp(mdp, horizon=500)
```
Comment (elliottower) [inline src/overcooked_ai_py/mdp/overcooked_env.py:696]: Ah I see there’s a horizon argument so maybe that’s the done signal you used earlier for truncation

File: testing/overcooked_test.py [hunk 4]
```diff
@@ -1714,6 +1714,27 @@ def test_creation(self):
...
+        agent_dir = os.path.join(
+            current_dir,
+            "../src/overcooked_demo/server/static/assets/agents/RllibCrampedRoomSP/agent",
+        )
+        ap = load_agent_pair(agent_dir, "ppo", "ppo")
+        env = OvercookedEnv.from_mdp(base_mdp, info_level=0, horizon=1000)
+        from overcooked_ai_py.mdp.overcooked_env import OvercookedEnv_Pet
+
+        wrapped_env = OvercookedEnv_Pet(env, ap)
+        parallel_api_test(wrapped_env, num_cycles=1000)
```
Comment (elliottower) [inline testing/overcooked_test.py:1735]: Looks good to me

**Comment by elliottower** [comment]
Left some comments, don’t have a chance to run through the code tho

File: src/overcooked_ai_py/mdp/overcooked_env.py (hunk 1 above)
Comment (jyan1999) [reply src/overcooked_ai_py/mdp/overcooked_env.py:678]: Yeah that's correct. The goal for agents in this environment is to obtain as much reward as possible under a fixed amount of time, so there is no early-termination condition and there is only the time limit. 

Just one other thing, this class (OvercookedEnv_PettingZoo) is not actually used and was a leftover when I was playing around with PettingZoo. The actual class is the one below (OvercookedEnv_Pet).

File: src/overcooked_ai_py/mdp/overcooked_env.py (hunk 2 above)
Comment (jyan1999) [reply src/overcooked_ai_py/mdp/overcooked_env.py:687]: thanks! I will update this and will keep this in mind in the future

File: src/overcooked_ai_py/mdp/overcooked_env.py (hunk 3 above)
Comment (jyan1999) [reply src/overcooked_ai_py/mdp/overcooked_env.py:696]: Yeah that's correct

**Comment by elliottower** [comment]
Any update on this? Looks like it hasn't been worked on in a few weeks. Can help out reviewing more code if need be

**Comment by micahcarroll** [comment]
@jyan1999 is not actively working on this project anymore, so things may be a bit slower. @jyan1999, do you expect to wrap this up / get round to this anytime soon? Otherwise, I can step in! Just let me know 😄

**Comment by jyan1999** [comment]
Yes, sorry! I have been busy traveling for the past 2 weeks. I do plan to wrap this up in the next day or two.

**Comment by jyan1999** [comment]
Hi @elliottower I have made some updates and addressed your comments. Let me know if there is anything missing and I will be happy to look into them.

File: src/overcooked_ai_py/mdp/overcooked_env.py [hunk 5]
```diff
@@ -772,7 +772,7 @@ def reset(self, seed=None, options=None):
             agent: self.agent_map[agent].featurize(dummy_state)[0]
             for agent in self.agents
         }
-        return obs_dict
+        return obs_dict, None
```
Comment (elliottower) [inline src/overcooked_ai_py/mdp/overcooked_env.py:775]: Looks good to me

**Review by elliottower — APPROVED** [review]

**Comment by elliottower** [comment]
Cool to see it passing the workflow now, I’ll update the third party environments listing soon once this is merged. Just FYI don’t remember which version of PZ this was but 1.23.1 is out and is a slight bug fix from 1.23.0 (which was the API change with returning obs, info)

**Comment by jyan1999** [comment]
> Cool to see it passing the workflow now, I’ll update the third party environments listing soon once this is merged. Just FYI don’t remember which version of PZ this was but 1.23.1 is out and is a slight bug fix from 1.23.0 (which was the API change with returning obs, info)

Yes the current version is working with 1.23.1. The API change was what was causing the workflow to fail.

**Pull request merged**